### PR TITLE
New module: fasta_windows

### PR DESCRIPTION
**Pull request by muffato**
`fasta_windows` is a tool to quickly compute statistics over a fasta file in windows.
https://github.com/tolkit/fasta_windows

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/fastawindows/main.nf [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    rm -rf fw_out
+    env RAYON_NUM_THREADS=$task.cpus \\
+    fasta_windows \\
+        $args \\
+        --fasta $fasta \\
+        --output ${prefix}_fw
```
Comment (jfy133) [inline modules/fastawindows/main.nf:33]:
```suggestion
        --output ${prefix}
```

Should be updated elsewhere as well, please see [^1] point 5 if you fear a clash in file names etc.

[^1]: https://nf-co.re/docs/contributing/modules#naming-conventions

File: modules/fastawindows/meta.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ id:'test', single_end:false ]
+  - fasta:
+      type: file
+      description: FASTA file
+      pattern: "*.{fa}"
```
Comment (jfy133) [inline modules/fastawindows/meta.yml:24]:
```suggestion
      pattern: "*.{fa,fasta,fna}"
```

**Review by jfy133 — CHANGES_REQUESTED** [review]
Output name shouldn't include anything other than prefix (up to pipeline developer to decide this), otherwise LGTM

**Comment by muffato** [comment]
Both fixed

**Review by jfy133 — APPROVED** [review]

**Comment by muffato** [comment]
Thank you James !

**Pull request merged**